### DISCUSSION TITLE
Does Capictor Prefreses work diferently in the web on firefox or chrome than in a emultor for IOS or Android?

### DISCUSSION THREAD

**Discussion by Hude06** [ionic-team/capacitor#7703] [Q&A]
Hey guys, I'm working on a project where I need to save a private key and some personal info on the users phone. I'm beta testing in the web using just plain HTML and JS and I have linked the capacitor API's. In the web at least it does not seem to save the data for more than about one day and when I get into the data tab of the console it seems like it is just saving it in the local storage whitch is what I don't want to happen. I need it to be permanent and I need to be able to trust that it will always be there

**Comment by jcesarmobile** [comment]
The preferences plugin uses localStorage on web, on Android uses SharedPreferences, and on iOS it uses UserDefaults.
It's explained on the docs
https://capacitorjs.com/docs/apis/preferences

While those should be persistent, they are not secure for storing private keys, for sensitive information, plugins such as identity vault are recommended, but it's a paid plugin
https://ionic.io/docs/identity-vault

you can search for free alternatives that use keychain on iOS and whatever secure storage android provides.
